docs/static/persistent-queues.asciidoc

```diff
@@ -128,22 +131,35 @@ events.
 [[durability-persistent-queues]]
 ==== Controlling Durability
 
+// I'm not sure about this particular description of durability. Might need some help.
+Durability is a property of storage writes that indicate the availability of
+that data after it was written.
+
 When the persistent queue feature is enabled, Logstash will store events on
-disk. The persistent queue exposes the trade-off between performance and
-durability by providing the following configuration options:
+disk. Logstash commits to disk in a mechanism we call checkpointing.
+
+To discuss durability, we need to introduce a few details about how the persistent queue is implemented.
+
+First, the queue itself is a set of pages. There are two kinds of pages: head page and tail page. The head page is where new events are written. When this head page is of a certain size (see `queue.page_capacity), the becomes a tail page, and a new head page is created. Tail pages are immutable. 
+
+Second, the queue tracks details about the queue (pages, acknowledgements, etc) in a separate file called a checkpoint file.
+
+When Logstash does checkpoint, the following will occur:
+
+* The head page is sync'd to disk.
+* A new checkpoint is written atomically to persist the current state of the queue.
+
+The following settings are available to let you tune durability:
+
+* `queue.checkpoint.writes`: The number of writes from inputs after which a checkpoint is written. 
```

Comment:
This description is a bit confusing because of the sentence structure. Does this say what you meant to say?

"The number of times an input writes to the queue before a checkpoint is written?"